screens/HomeScreen.js

```diff
@@ -1,45 +1,105 @@
 import React from 'react';
-import {
-  AsyncStorage,
-  Button,
-  Image,
+import { 
+  AsyncStorage, 
+  Button, 
+  Image, 
   Platform,
   ScrollView,
   StyleSheet,
   Text,
-  View
+  View,
+  RefreshControl
 } from 'react-native';
-
 import { MonoText } from '../components/StyledText';
 import BASE from '../lib/common';
 
 export default class HomeScreen extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      userId: '',
       name: '',
-      points: ''
+      points: '',
+      rewards: [],
+      redeemable: {},
+      announcements: '',
+      transactions: [],
+      latestTransaction: '',
+      refreshing: false, 
+      updates: false
     };
   }
 
+  // This is what runs when you pull to refresh. It is essentially the
```

Comment:
Curious if we're able to break part of it into a helper function?

Reply:
Yep